datafusion/proto/src/from_proto.rs

```diff
@@ -656,21 +658,22 @@ impl TryFrom<&protobuf::ScalarListValue> for ScalarValue {
     fn try_from(
         scalar_list_value: &protobuf::ScalarListValue,
     ) -> Result<Self, Self::Error> {
-        use protobuf::{scalar_type::Datatype, PrimitiveScalarType};
+        if 2 > 1 {
```

Comment:
@tustvold im not sure why all tests are ok, looks like this piece of code not used. The CI build also has Ballista tests?

Reply:
Maybe it is because the tests don't have coverage of `DataType::List` 

This is a good experiment (add in a `panic`) but we probably need to remove this prior to merging this PR

Reply:
Right,I added this to check if this code part is used. We have roundtrips tests for lists like `round_trip_scalar_values` which passed. I'm wondering if ballista tests are in CI? Otherwise this code looks useless and can be dropped

Reply:
I think the ballista CI checks only run in the ballista repo 🤔